### PR TITLE
Incubator.TextField missing features and fixes

### PR DESCRIPTION
**Pull request by ethanshar**
## Description
Fix and add missing features
- Fix issue with floatingPlaceholder doesn't retain space for the floating state
- Modifiers support
- Replace leading/trailingButton prop with a more general API -> leading/trailingAccessory.

## Changelog
Incubator.TextField fixes and features: 
- Fix issue with floatingPlaceholder doesn't retain space for the floating state
- Add Modifiers support (margins, padding, typography and color) 
- Replace leading/trailingButton prop with a more general API -> leading/trailingAccessory that can accept any element.

### CODE REVIEW

**Review by M-i-k-e-l — COMMENTED** [review]
What was this issue:
> Fix issue with floatingPlaceholder doesn't retain space for the floating state

File: src/incubator/TextField/Label.tsx [hunk 1]
```diff
@@ -54,5 +54,11 @@ const Label = ({
   return null;
 };
 
+const styles = StyleSheet.create({
+  dummyPlaceholder: {
+    opacity: 0
+  }
+});
+
 Label.displayName = 'Incubator.TextField';
```
[the remark this answers is not in the document]
Comment (ethanshar) [reply src/incubator/TextField/Label.tsx:63]: Yes, there's a new logic that unify all internal components into a single component in the docs (if they have the same displayName)
This is useful in cases you want to break a big components into smaller parts but still want to have a single page for the component. 

Incubator is the only parent that doesn't have an entry cause it's really a namespace and not a component
Maybe we can create a dedicate page for it, that explain what's in Incubator. 
(I'll do it as a separate task)

File: demo/src/screens/componentScreens/IncubatorTextFieldScreen.tsx [hunk 2]
```diff
@@ -118,8 +155,8 @@ export default class TextFieldScreen extends Component {
               Colors By State
             </Text>
             <Button
-              label={shouldDisable ? 'Enable' : 'Disable'} 
-              onPress={() => this.setState({shouldDisable: !shouldDisable})} 
+              label={shouldDisable ? 'Enable' : 'Disable'}
+              onPress={() => this.setState({shouldDisable: !shouldDisable})}
```
[the remark this answers is not in the document]
Comment (ethanshar) [reply demo/src/screens/componentScreens/IncubatorTextFieldScreen.tsx:160]: 1. Fixed 
2. Done 👍

**Comment by ethanshar** [comment]
> What was this issue:
> 
> > Fix issue with floatingPlaceholder doesn't retain space for the floating state

Because floating placeholder is a fake placeholder and it has absolute position it doesn't take actual space when rendered. 
So there was an issue that when you focus on an input and the floating placeholder goes up it can overlap with other elements if there's not enough space. 

So in order to avoid that we needed to retain some empty space for the floating placeholder to float to (:

**Review by M-i-k-e-l — COMMENTED** [review]
Two more bugs (Android only):
1. Accessories - the `ActivityIndicator` is not visible on Android; add `color={Constants.isIOS ? Colors.dark60 : Colors.blue40}` to the `ActivityIndicator`.
2. In the Enable\Disable TextField: add text --> press the disable button.
    Result: the text disappears.

**Comment by ethanshar** [comment]
> Two more bugs (Android only):
> 
> 1. Accessories - the `ActivityIndicator` is not visible on Android; add `color={Constants.isIOS ? Colors.dark60 : Colors.blue40}` to the `ActivityIndicator`.
> 2. In the Enable\Disable TextField: add text --> press the disable button.
>    Result: the text disappears.

1. Fixed the first issue. 
2. This issue seems like a react-native issue. It happens for React Native TextInput. 
Apparently when changing the style of the input color (default color is grey10 and when it's disabled it changes to grey40) it just clears the value which is very strange. 
I will open a separate ticket for it cause I don't won't it to block this PR.

**Review by M-i-k-e-l — APPROVED** [review]

**Pull request merged**